### PR TITLE
Fix `messages` argument typing for `delete_messages` API

### PR DESCRIPTION
**Pull request by davfsa**

### CODE REVIEW

File: hikari/api/rest.py [hunk 1]
```diff
@@ -1455,9 +1456,9 @@ async def delete_messages(
         channel : hikari.snowflakes.SnowflakeishOr[hikari.channels.TextableChannel]
             The channel to bulk delete the messages in. This may be
             the object or the ID of an existing channel.
-        messages : typing.Union[hikari.snowflakes.SnowflakeishOr[hikari.messages.PartialMessage], hikari.snowflakes.SnowflakeishIterable[hikari.messages.PartialMessage]]
+        messages
```
Comment (GoogleGenius) [inline hikari/api/rest.py:1459]: Is the removal intentional?
```suggestion
        messages : typing.Union[snowflakes.SnowflakeishOr[messages_.PartialMessage], typing.Iterable[snowflakes.SnowflakeishOr[messages_.PartialMessage]], typing.AsyncIterable[snowflakes.SnowflakeishOr[messages_.PartialMessage]]],
```

Comment (davfsa) [reply hikari/api/rest.py:1459]: Yes. The type would become so big that it would look back both in the file and in the documentation

**Review by FasterSpeeding — DISMISSED** [review]

**Comment by FasterSpeeding** [comment]
You need to update https://github.com/hikari-py/hikari/blob/ef7bdc767e1db2e078f54455d18e30aac74563dc/hikari/channels.py#L778

**Comment by davfsa** [comment]
> You need to update
> 
> https://github.com/hikari-py/hikari/blob/ef7bdc767e1db2e078f54455d18e30aac74563dc/hikari/channels.py#L778

Nice spot, thanks!

Sorted out :)

**Pull request merged**